lib/foreman_tasks/cleaner.rb

```diff
@@ -90,13 +93,29 @@ def tasks
       ForemanTasks::Task.search_for(full_filter).select('DISTINCT foreman_tasks_tasks.id, foreman_tasks_tasks.type, foreman_tasks_tasks.external_id')
     end
 
-    def delete_tasks(chunk)
-      ForemanTasks::Task.where(:id => chunk.map(&:id)).delete_all
+    def delete_tasks(chunk, backup_dir = ForemanTasks.dynflow.world.persistence.current_backup_dir)
```

Comment:
It seems to me having the `backup_dir` as instance variable configured in the initializer would help here, wouldn't it? It would also reduce some code duplicity.